app/models/thredded/messageboard.rb

```diff
@@ -128,8 +128,13 @@ def unread_topics_counts(user:, topics_scope: Thredded::Topic.all)
         messageboards = arel_table
         read_states = Thredded::UserTopicReadState.arel_table
         topics = topics_scope.arel_table
-        joins(:topics).merge(topics_scope).joins(
-          messageboards.outer_join(read_states).on(
+        topics_ids = topics_scope.pluck(:id)
```

Comment:
This is now 2 queries which is not ideal.

I'd prefer that we figure out why it stopped working in Rails 6.2.rc2 and either file a bug in Rails or fix it on our side properly.

Reply:
Filed a bug in Rails